### ISSUE TITLE
Feature Request: Change styles for different levels while indenting bullet/numbered lists

### ISSUE THREAD

**Issue by g105b** [nylas/nylas-mail#1541]
Operating system: Ubuntu Linux 15.10
N1 version: 0.4.10
No third-party plugins installed.

This issue is related to the email text editor. 

In most word processors or other rich-text-capable editors, when working with bulleted or numbered lists, indenting causes the lists to iterate through different styles.

---

Expected behaviour when indenting lists:

```
• This is a bulleted list
• Another bullet here
    ◦ When indenting, bullet design changes.
    ◦ This is just a nice feature with bullets.
        ⬛ But it is an important feature when working with numeric lists.
        ⬛ See below.

1. The reason it is important with numbered lists.
2. Is because when referring back to a numbered list...
3. ... it is very useful to be able to quote the number.
    a. So, when there is an indented list...
    b. It makes sense to use a different counter scheme.
        i. This is how most word processors do it
        ii. And how most email clients do it.
```

Actual behaviour in N1:

```
• This is a bulleted list
• Another bullet here
    • When indenting, bullet design changes.
    • This is just a nice feature with bullets.
        • But it is an important feature when working with numeric lists.
        • See below.

1. The reason it is important with numbered lists.
2. Is because when referring back to a numbered list...
3. ... it is very useful to be able to quote the number.
    1. So, when there is an indented list...
    2. It makes sense to use a different counter scheme.
        1. This is how most word processors do it
        2. And how most email clients do it.
```

As you can see, it would be very useful to have numeric lists behave in a way that allows sub lists to be identified by number-alpha-roman style.

**Comment by g105b** [comment]
Regarding how this can be achieved, I believe that this can be done in CSS alone, using the `counter()` method.

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Lists_and_Counters/Using_CSS_counters

**Comment by bengotow** [comment]
Hey! It'd be really interesting to see if we can do this in CSS, but I'm not sure if we could "bake" the CSS into text before we send the email? I think we'd want to avoid relying solely on the CSS since the recipient's client may not support `counter()`.

This is our code for the list behavior currently: https://github.com/nylas/N1/blob/master/src/components/contenteditable/list-manager.coffee#L105.

I think it eventually calls through to `insertOrderedList` and `insertUnorderedList`, which (believe it or not) are actually built in to `contenteditable`: https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

**Comment by g105b** [comment]
That's a good point, I hadn't thought of the rendering side of things.

I have certainly achieved this with CSS only before, but as you say, this
only solves the problem for the composer, not the recipient.

The question comes down to whether it makes any sense to manipulate the
lists when sending, rather than sending standard LI tags in the HTML... If
not, I guess this issue should be closed.

**Comment by emorikawa** [comment]
@g105b Really cool idea about using CSS counters

It looks like basic CSS counters have pretty good browser-compatibility: http://caniuse.com/#search=counter so I think we may indeed be able to bake it into the text before sending. Since the fallback case is what we have right now, I definitely say for it.

It may not even be that bad to implement. If you take a look at src/components/contenteditable/list-manager.coffee you'll see a method called `onContentChanged` This method gets called every time the composer's DOM changes (via a MutationListener) and lets you fix whatever you want via raw DOM methods. You can see we do stuff like concatenate adjacent lists and other cleanup in there. `editor.rootNode` is the raw, queryable, mutable DOM of the current composer.

Basically I think you can make one little function that looks for nested list items, and sets some inline styles (if they doesn't exist already) to have the appropriate list indentation.

The main trick might be to see if this is possible with inline styles alone. While we theoretically could append a `<style>` tag, those tend to get stripped or inlined anyway by most email clients.

**Comment by g105b** [comment]
I'm using Mailspring now and this is partially implemented.
